clientconn.go

```diff
@@ -411,6 +419,7 @@ func (cc *ClientConn) Close() error {
 	}
 	cc.state = Shutdown
 	cc.stateCV.Broadcast()
+	cc.events.Finish()
```

Comment:
Finish should only be called once.  The return at line 418 happens when Close has already been called; we don't want to call Finish in that case.